### PR TITLE
Sync activities

### PR DESCRIPTION
**Pull request by Claudio5**
This pull request tries to synchronize the different activities
- ChooseWordsActivity
- DrawingActivity
- VotingActivity

For the moment the user gets in the room and needs to wait for a second player. Once a second player enters a timer of 10 seconds (test time) starts and afterwards the drawing activity starts. Once the drawing sctivity starts players have limited time to draw (for testing I just put 15 seconds). Then it redirects to voting activity. 

N.B. The syncronization between the drawings and voting page still needs to be done

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/SDP/matchmaking/GameStates.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package ch.epfl.sweng.SDP.matchmaking;
+
+public class GameStates {
+    public final static int HOMESTATE = 0;
+    public final static int CHOOSE_WORDS_TIMER_START = 1;
+    public final static int START_DRAWING_ACTIVITY = 2;
+    public final static int START_VOTING_ACTIVITY = 3;
+    public final static int END_VOTING_ACTIVITY = 4;
+}
```
Comment (LudoHoff) [inline app/src/main/java/ch/epfl/sweng/SDP/matchmaking/GameStates.java]: Replace by an enum ?

File: app/src/main/java/ch/epfl/sweng/SDP/BooleanVariableListener.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package ch.epfl.sweng.SDP;
+
+public class BooleanVariableListener {
+    private boolean boo;
```
Comment (Gondolav) [inline app/src/main/java/ch/epfl/sweng/SDP/BooleanVariableListener.java]: bool is better ;)

File: app/src/main/java/ch/epfl/sweng/SDP/game/VotingPageActivity.java [hunk 3]
```diff
@@ -36,15 +43,64 @@
...
         @Override
         public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
-            Integer value = dataSnapshot.getValue(Integer.class);
+            Integer state = dataSnapshot.getValue(Integer.class);
+
+            if(state != null) {
+                GameStates stateEnum = GameStates.convertValueIntoState(state);
+                switch(stateEnum) {
+                    case END_VOTING_ACTIVITY:
+                        // Start ranking activity
```
Comment (Gondolav) [inline app/src/main/java/ch/epfl/sweng/SDP/game/VotingPageActivity.java]: You have to use here `
// Prepare a Bundle for passing the ranking array to the fragment
                Bundle bundle = new Bundle();
                bundle.putStringArray("Ranking", ranking);

                // Clear the UI; buttonChangeImage and rankingButton need
                // to be removed after testing
                setVisibility(View.GONE, R.id.ratingBar, R.id.drawing, R.id.playerNameView,
                        R.id.buttonChangeImage, R.id.rankingButton);

                // Create and show the final ranking in the new fragment
                getSupportFragmentManager().beginTransaction()
                        .add(R.id.votingPageLayout,
                                RankingFragment.instantiate(getApplicationContext(),
                                        RankingFragment.class.getName(), bundle))
                        .addToBackStack(null).commit();`

Comment (Claudio5) [reply app/src/main/java/ch/epfl/sweng/SDP/game/VotingPageActivity.java]: Will be implemented in a future pull request

**Pull request merged**